surefire-providers/surefire-junit-platform/src/test/java/org/apache/maven/surefire/junitplatform/RunListenerAdapterTest.java

```diff
@@ -425,7 +425,7 @@ public void notifiedOfContainerFailure()
                     throws Exception
     {
         adapter.executionFinished( newContainerIdentifier(), failed( new RuntimeException() ) );
-        verify( listener ).testFailed( any() );
+        verify( listener ).testError( any() );
```

Comment:
You are modifying the old tests?
The reason between error and failure is that failure throws `AssertionError` but the error throws any other exception.

Reply:
Right. I initially just re-used the existing logical flow when I made my fix, which led to it being a failure, not an error. So I changed that in this latest set of changes. I can add a check [like the one a few lines below my changes](https://github.com/apache/maven-surefire/pull/267/commits/6df45c50e06b396504b0a846a59a60052f0177ff#diff-ad07b0c74885ecb3d687dcc85495e390R136-R137) to see whether or not it's an `AssertionError` if you'd prefer. 

I was actually planning on adding that check for my latest commit, but noticed that the `isClass` (previously `!isTest`) doesn't check that, so instead I limited the scope of my changes and decided I'd just leave it for another PR. 

`isClass` block:
https://github.com/apache/maven-surefire/blob/6df45c50e06b396504b0a846a59a60052f0177ff/surefire-providers/surefire-junit-platform/src/main/java/org/apache/maven/surefire/junitplatform/RunListenerAdapter.java#L121-L126

If I do add that `AssertionError` check, should I add it to the `isClass` block as well? Or just the `isContainer` block?

Reply:
I see you crated the test methods in the commit `136f79ee`. So it's your's, just fine when you modify your's!

Feel free to rename `!isTest` to `isClass`. I don't have a problem with that due to you have extended the branching with `else if ( testIdentifier.isContainer() )`. If the new argument with `isClass` is fully logical and right to do, then use it of course here.

Reply:
Ok, this whole case has been updated to reflect the changes I was suggesting. Test failures and errors should be properly respected for every `FAILURE` now (previously a `RuntimeException` would not have been considered a test error for `@BeforeAll`, for example, it would instead be treated as a failure).